Makefile

```diff
@@ -96,11 +96,11 @@ libzim-asm.dev.js: libzim_bindings.cpp prejs_file_api.js postjs_file_api.js
 
 # Production WASM version with WORKERFS and NODEFS, optimized and packed
 libzim-wasm.js: libzim_bindings.cpp prejs_file_api.js postjs_file_api.js
-	em++ -o libzim-wasm.js --bind libzim_bindings.cpp -I/src/build/include -L/src/build/lib -lzim -llzma -lzstd -lxapian -lz -licui18n -lpthread -licuuc -licudata -O3 --pre-js prejs_file_api.js --post-js postjs_file_api.js -s WASM=1 -s "EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','err','ALLOC_STACK','out']" -s INITIAL_MEMORY=83886080 -s ALLOW_MEMORY_GROWTH=1 -s DYNAMIC_EXECUTION=0 -lworkerfs.js -lnodefs.js
+	em++ -o libzim-wasm.js --bind libzim_bindings.cpp -I/src/build/include -L/src/build/lib -lzim -llzma -lzstd -lxapian -lz -licui18n -lpthread -licuuc -licudata -O3 --pre-js prejs_file_api.js --post-js postjs_file_api.js -s WASM=1 -s "EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','err','ALLOC_STACK','out']" -s INITIAL_MEMORY=83886080 -s EXCEPTION_CATCHING_ALLOWED=["getEntryByPath","search","getArticleCount","loadArchive"] -s ALLOW_MEMORY_GROWTH=1 -s DYNAMIC_EXECUTION=0 -lworkerfs.js -lnodefs.js
 
```

Comment:
I presume this type of exception catching need to be reverted to the catch-all type.